export_dagmc_cmd/DAGMCExportCommand.cpp

```diff
@@ -0,0 +1,856 @@
+          message << "Surface " << face->id() << " has reverse sense " <<
+            "with multiple volume " << reverse->id() << " and " <<
+            "volume " << vol->id() << std::endl;
+          return moab::MB_FAILURE;
+        }
+        reverse = vol;
+      }
+      if (cf->get_sense() == CUBIT_UNKNOWN ||
+          cf->get_sense() == face->get_surface_ptr()->bridge_sense()) {
+        // Check that each surface has a sense for only one volume
+        if (forward) {
+          message << "Surface " << face->id() << " has forward sense " <<
+            "with multiple volume " << forward->id() << " and " <<
+            "volume " << vol->id() << std::endl;
+          return moab::MB_FAILURE;
+        }
+        forward = vol;
+      }
+    }
+
+    if (forward) {
+      rval = myGeomTool->set_sense(ci->second, volume_map[forward], moab::SENSE_FORWARD);
+      if (moab::MB_SUCCESS != rval) return rval;
+    }
+    if (reverse) {
+      rval = myGeomTool->set_sense(ci->second, volume_map[reverse], moab::SENSE_REVERSE);
+      if (moab::MB_SUCCESS != rval) return rval;
+    }
+  }
+
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode DAGMCExportCommand::store_curve_senses(refentity_handle_map& curve_map,
+                                      refentity_handle_map& surface_map)
+{
+  moab::ErrorCode rval;
+  std::vector<moab::EntityHandle> ents;
+  std::vector<int> senses;
+  refentity_handle_map_itor ci;
+  for (ci = curve_map.begin(); ci != curve_map.end(); ++ci) {
+    RefEdge* edge = (RefEdge*)(ci->first);
+    ents.clear();
+    senses.clear();
+    for (SenseEntity* ce = edge->get_first_sense_entity_ptr();
+         ce; ce = ce->next_on_bte()) {
+      BasicTopologyEntity* fac = ce->get_parent_basic_topology_entity_ptr();
+      moab::EntityHandle face = surface_map[fac];
+      if (ce->get_sense() == CUBIT_UNKNOWN ||
+          ce->get_sense() != edge->get_curve_ptr()->bridge_sense()) {
+        ents.push_back(face);
+        senses.push_back(moab::SENSE_REVERSE);
+      }
+      if (ce->get_sense() == CUBIT_UNKNOWN ||
+          ce->get_sense() == edge->get_curve_ptr()->bridge_sense()) {
+        ents.push_back(face);
+        senses.push_back(moab::SENSE_FORWARD);
+      }
+    }
+
+    rval = myGeomTool->set_senses(ci->second, ents, senses);
+    if (moab::MB_SUCCESS != rval) return rval;
+  }
+  return moab::MB_SUCCESS;
+}
+
+
+moab::ErrorCode DAGMCExportCommand::store_groups(refentity_handle_map (&entitymap)[5])
+{
+  moab::ErrorCode rval;
+
+  // Create entity sets for all ref groups
+  rval = create_group_entsets(entitymap[4]);
+  if (moab::MB_SUCCESS != rval) return rval;
+
+  // Store group names and entities in the mesh
+  rval = store_group_content(entitymap);
+  if (moab::MB_SUCCESS != rval) return rval;
+
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode DAGMCExportCommand::create_group_entsets(refentity_handle_map& group_map)
+{
+  moab::ErrorCode rval;
+  const char geom_categories[][CATEGORY_TAG_SIZE] =
```

Comment:
This structure is defined in multiple places. Could probably be make a static array in the header.